### PR TITLE
Replace react icons with lucidedev

### LINKED ISSUE #7959: [CAL-1349] Replace React-Icons with Lucide.dev
We really only use Feather Icons from React-Icons and Lucide.dev is the successor of React Feathers.

while there is a PR open to add Lucide to React Icons, it's been open since September 2022 and I lost faith it'll get merged anytime soon.

https://github.com/react-icons/react-icons/pull/602

Is there any objections replacing react-icons with lucide.dev's react implementation? https://lucide.dev/docs/lucide-react

<sub>[CAL-1349](https://linear.app/calcom/issue/CAL-1349/replace-react-icons-with-lucidedev)</sub>

### PR DESCRIPTION
**Pull request by abdallahalsamman**
[Previous PR](https://github.com/calcom/cal.com/pull/8072) got closed because it was by no way complete.

## What does this PR do?

Add lucide-dev package and replaced webhook icon to use `lucide-react` instead of `react-icons/tb` and `react-icons/fi`

Fixes #7959 

## Note

Migrating away from `react-icons/fa` and `react-icons/go` remains. these have no alternative icons in `lucide-react`. maybe we keep them this way? or should I add an svg for both?

### CODE REVIEW

**Comment by PeerRich** [comment]
great work! will test today!

File: apps/web/next.config.js [hunk 1]
```diff
@@ -94,11 +94,11 @@ const nextConfig = {
...
-    "@calcom/ui/components/icon": {
-      transform: "@react-icons/all-files/fi/{{member}}",
-      skipDefaultConversion: true,
-      preventFullImport: true,
-    },
+    // "@calcom/ui/components/icon": {
+    //   transform: "@react-icons/all-files/fi/{{member}}",
+    //   skipDefaultConversion: true,
+    //   preventFullImport: true,
+    // },
```
Comment (PeerRich) [inline apps/web/next.config.js:101]: is this a good idea @zomars?

Comment (PeerRich) [reply apps/web/next.config.js:101]: @abdallahalsamman why was this removed? i think we can keep the transformer, we just need to change the import schema

Comment (abdallahalsamman) [reply apps/web/next.config.js]: @PeerRich should be fixed :)

**Comment by PeerRich** [comment]
CleanShot 2023-04-09 at 00 17 59@2x
@abdallahalsamman getting an error here

**Comment by abdallahalsamman** [comment]
On it

**Comment by abdallahalsamman** [comment]
@PeerRich can you try now please? I made sure `yarn build` passes the eslinting and creating an optimized build steps.

**Comment by PeerRich** [comment]
looks good now, however there are some merge conflicts now

**Comment by abdallahalsamman** [comment]
@PeerRich merge conflicts fixed

**Review by emrysal — APPROVED** [review]
Thanks you for the contribution and also the back-and-forth amends over Slack, LGTM 🚀

**Comment by zomars** [comment]
A customer reported an error on their CI with this change. Will post logs in a bit.

**Comment by zomars** [comment]
This is what's happening. Already tried to revert, hence the `FiX` instead of `X`.

image

**Comment by zomars** [comment]
@calcom/foundation seems like the OOM issue is coming from here. Will report back once I get more info.

**Pull request merged**